pkg/containerutil/containerutil.go

```diff
@@ -68,3 +72,80 @@ func ContainerStatus(ctx context.Context, c containerd.Container) (containerd.St
 
 	return task.Status(ctx)
 }
+
+func withOSEnv(envs []string) ([]string, error) {
+	newEnvs := make([]string, len(envs))
+
+	// from https://github.com/docker/cli/blob/v22.06.0-beta.0/opts/env.go#L18
+	getEnv := func(val string) (string, error) {
+		arr := strings.SplitN(val, "=", 2)
+		if arr[0] == "" {
+			return "", errors.New("invalid environment variable: " + val)
+		}
+		if len(arr) > 1 {
+			return val, nil
+		}
+		if envVal, ok := os.LookupEnv(arr[0]); ok {
+			return arr[0] + "=" + envVal, nil
+		}
+		return val, nil
+	}
+	for i := range envs {
+		env, err := getEnv(envs[i])
+		if err != nil {
+			return nil, err
+		}
+		newEnvs[i] = env
+	}
+
+	return newEnvs, nil
+}
+
+func parseEnvVars(paths []string) ([]string, error) {
+	vars := make([]string, 0)
+	for _, path := range paths {
+		f, err := os.Open(path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open env file %s: %w", path, err)
+		}
+		defer f.Close()
+
+		sc := bufio.NewScanner(f)
+		for sc.Scan() {
+			line := strings.TrimSpace(sc.Text())
+			// skip comment lines
+			if strings.HasPrefix(line, "#") {
+				continue
+			}
+			vars = append(vars, line)
+		}
+		if err = sc.Err(); err != nil {
+			return nil, err
+		}
+	}
+	return vars, nil
+}
+
+// GenerateEnvs combines environment variables from `--env-file` and `--env`.
+// Pass an empty slice if any arg is not used.
+func GenerateEnvs(envFile []string, env []string) ([]string, error) {
```

Comment:
Here might be a better place for these 3 funcs. It'll be good to add some unittests for `GenerateEnvs`. (It's okay to not add in this PR, but please open an issue :))

https://github.com/containerd/nerdctl/blob/c2c648c5ef7bad5ebbcb05dcdef9247f31892efe/pkg/flagutil/flagutil.go

Reply:
I see. I'll do it in another PR.

Reply:
I mean you can move the 3 funcs to `flagutil` (instead of `containerutil`) in this PR, and create an issue to track adding unittest for it/`GenerateEnvs` :)

Reply:
Make sense. :) 

I have opened the issue. #1884

Reply:
Functions moved. Done.